tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,299 @@
+"""
+Tests tagging rest api views
+"""
+
+import ddt
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from openedx_tagging.core.tagging.models import Taxonomy
+
+User = get_user_model()
+
+
+
+@ddt.ddt
+class TestTaxonomyViewSet(APITestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.user = User.objects.create(
+            username="user",
+            email="user@example.com",
+        )
+
+        self.staff = User.objects.create(
+            username="staff",
+            email="staff@example.com",
+            is_staff=True,
+        )
+
+    @ddt.data(
+        (None, status.HTTP_200_OK),
+        (1, status.HTTP_200_OK),
+        (0, status.HTTP_200_OK),
+        (True, status.HTTP_200_OK),
+        (False, status.HTTP_200_OK),
+        ("True", status.HTTP_200_OK),
+        ("False", status.HTTP_200_OK),
+        ("1", status.HTTP_200_OK),
+        ("0", status.HTTP_200_OK),
+        (2, status.HTTP_400_BAD_REQUEST),
+        ("invalid", status.HTTP_400_BAD_REQUEST),
+    )
+    @ddt.unpack
+    def test_list_taxonomy_queryparams(self, enabled, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
+        Taxonomy.objects.create(name="Taxonomy enabled 1", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy enabled 2", enabled=True).save()
+        Taxonomy.objects.create(name="Taxonomy disabled", enabled=False).save()
+
+        self.client.force_authenticate(user=self.staff)
+        if enabled is not None:
+            response = self.client.get(url, {"enabled": enabled})
+        else:
+            response = self.client.get(url)
+        self.assertEqual(response.status_code, expected_status)
+
+        # If we were able to create the taxonomy, check if it was created
+        if status.is_success(expected_status):
+            if enabled is None:
+                self.assertEqual(len(response.data), 3)
+            elif enabled in [True, "True", 1, "1"]:
+                self.assertEqual(len(response.data), 2)
+            else:
+                self.assertEqual(len(response.data), 1)
+
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_403_FORBIDDEN),
+        ("staff", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_list_taxonomy(self, user_attr, expected_status):
+        url = reverse("oel_tagging:taxonomy-list")
+
+        user = getattr(self, user_attr) if user_attr else None
+        if user:
+            self.client.force_authenticate(user=user)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, expected_status)
+
+    @ddt.data(
+        (None, {"enabled": True}, status.HTTP_403_FORBIDDEN),
+        ("user", {"enabled": True}, status.HTTP_200_OK),
+        ("user", {"enabled": False}, status.HTTP_404_NOT_FOUND),
```

Comment:
Nope.. normal users aren't allowed to view taxonomies, only taxonomy admins.

When you remove `has_permission` as [suggested above](https://github.com/open-craft/openedx-learning/pull/4/files/a0fedd36bbe06ef0a4c00466c8c9570f3c35616d#diff-091bcd6aa37fc62100ec5e982af82be0886e0ea9025f8bc15bea40bd67f978adR21), this is reflected correctly, and should be:

```suggestion
        ("user", {"enabled": True}, status.HTTP_403_FORBIDDEN),
        ("user", {"enabled": False}, status.HTTP_403_FORBIDDEN),
```

Reply:
Commented above